dropshot/src/server.rs

```diff
@@ -200,8 +206,8 @@ impl<C: ServerContext> HttpServerStarter<C> {
             probe_registration,
             app_state: self.app_state,
             local_addr: self.local_addr,
-            join_handle: Some(join_handle),
-            close_channel: Some(tx),
+            closer: CloseHandle { close_channel: Some(tx) },
+            join_future: join_handle.boxed().shared(),
```

Comment:
This is neat.  I'd never run across `shared()` before.  (I usually wind up using a channel and cloning the receiver.)